src/liquid/components/ld-input/ld-input.css

```diff
@@ -468,8 +472,9 @@
   left: var(--ld-input-padding-x-md);
   margin-right: 0;
 
-  :host([disabled]) &,
-  .ld-input[disabled] & {
+  :host(.ld-input--disabled) &,
+  .ld-input:has(input:disabled) &,
```

Comment:
Do we really need both, and if yes, why only here?

Reply:
This makes sure both work, Web Component and CSS component. The CSS component has a wrapping container containing the input element. The container needs to have the style applied to it. Since the `:has` selector is still not supported in almost any browser right now, we must use this helper class unfortunately.